### ISSUE TITLE
version 0.1.0 to 0.3.0

### ISSUE THREAD

**Issue by cometta** [BerkeleyTrue/react-redux-epic#6]
I get below error, any example for using renderToString().subscribe(..) for 0.3.0? It was working on 0.1.0

```
/build/node_modules/rxjs/Subscriber.js:240
            throw err;
            ^

TypeError: Cannot set property 'length' of null
    at Subject.complete (/build/node_modules/rxjs/Subject.js:85:31)

```

**Comment by BerkeleyTrue** [comment]
@cometta Is there a stack tract that comes along with that error? It looks like it is only pointing to Rx code.

**Comment by BerkeleyTrue** [comment]
@cometta I'm able to replicate the error. Let me look into a fix

**Comment by BerkeleyTrue** [comment]
This one was a hard one to track down.

Looks like the behavior of Subject.prototype.complete was changed between Rx beta used in 0.2.0 and Rx v5 used in 0.3.0 so that it no longer creates a local var of the observers before eventually calling unsubscribe.

This causes a serious of events within renderToString which causes the error above:

1. [`wrappedRootEpic.complete`](https://github.com/BerkeleyTrue/react-redux-epic/blob/master/src/render-to-string.js#L44) is called in `initialRender`. 
2. This calls [`actionsProxy.complete`](https://github.com/BerkeleyTrue/react-redux-epic/blob/master/src/wrap-root-epic.js#L59) which intern causes the [`actions`](https://github.com/BerkeleyTrue/react-redux-epic/blob/master/src/wrap-root-epic.js#L34) obs' passed into the user epic to complete, which itself triggers the [`lifecycle.complete`](https://github.com/BerkeleyTrue/react-redux-epic/blob/master/src/wrap-root-epic.js#L42). This all happens synchronously.
3. `lifecycle` is a subject so it's complete will cycle through all the observers synchronously one by one and call their complete method [here](https://github.com/ReactiveX/rxjs/blob/master/src/Subject.ts#L91).
4. This synchronously triggers the `last` and `map` operators in [`renderToString`](https://github.com/BerkeleyTrue/react-redux-epic/blob/master/src/render-to-string.js#L50).
5. The `map` operator will call  [`wrappedRootEpic.unsubscribe`](https://github.com/BerkeleyTrue/react-redux-epic/blob/master/src/render-to-string.js#L50).
6. `wrappedRootEpic` will then call [`lifecycle.unsubscribe`](https://github.com/BerkeleyTrue/react-redux-epic/blob/master/src/wrap-root-epic.js#L63) which will null the observers property. Keep in mind that this is all happening synchronously which means that somewhere in the call stack the `lifecycle.complete` function has yet to resolve and is waiting for all of it's observers to complete. This is the last function on the call stack to finish. Once it resolves we start working our way back down the stack.
7. Once `lifecycle` observers have completed it will then set the length of the observers array to [zero](https://github.com/ReactiveX/rxjs/blob/master/src/Subject.ts#L93) (this effectively blanks our the array). But since the `lifecycle.unsubscribe` function was called before this function could  complete it will attempt to set the length property of a null object causing the error.


tl:dr; We need to make sure the `wrappedRootEpic.unsubscribe` happens after the `wrappedRootEpic.complete` function has resolved. 

I've got a fix for this but I want to get a failing test before I submit it.

**Comment by cometta** [comment]
if you create a branch of the changes, i can help you to test it.

**Comment by BerkeleyTrue** [comment]
@cometta I've got a PR going through travis if you want to check it out

**Comment by BerkeleyTrue** [comment]
Fix released as 0.3.1

**Comment by cometta** [comment]
yup, it's working in 0.3.1, thank you @BerkeleyTrue

**Comment by BerkeleyTrue** [comment]
👍
